packages/contracts/package.json

```diff
@@ -33,7 +33,7 @@
     "clean": "run-p clean:*",
     "clean:yarn": "yarn clean",
     "clean:forge": "forge clean",
-    "build": "forge build",
+    "build": "tsx ./scripts/prebuild/export-smtchecker-contracts.ts && forge build",
```

Comment:
You can also specify the order via `run-s build:smt build:compile`

Reply:
You don't have the `npm-run-all` dependency on that workspace, do you want me to add it? Apart from that, you don't have any tasks, you can only run `run-p build` or `run-s build`

Reply:
> don't have the `npm-run-all` dependency on that workspace

That's strange I didn't notice that before. Yeah I think adding it is fine!

> you don't have any tasks

My code suggestion is to add those two sub tasks so you can combine them with `run-s build:smt build:compile`

Reply:
I have basically zero formal Foundry experience so it's best if we get input from some of the team. @rndquu @zgorizzo69 @0xcodercrane

Reply:
what about creating a smt profile https://book.getfoundry.sh/config/?highlight=profile#configuring-with-foundrytoml so i.e 
if `FOUNDRY_PROFILE` is smtchecker then will run smt checker when running forge build otherwise it will not

Reply:
> what about creating a smt profile https://book.getfoundry.sh/config/?highlight=profile#configuring-with-foundrytoml so i.e if `FOUNDRY_PROFILE` is smtchecker then will run smt checker when running forge build otherwise it will not

Good point. @jpldcarvalho you could create 2 profiles: default (without smt checker) and the one which enables smt. Then run `forge build` using the smt profile.